modules/userdirectory/src/main/java/org/opencastproject/userdirectory/UserAndRoleDirectoryServiceImpl.java

```diff
@@ -340,7 +341,10 @@ private User loadUser(Tuple<String, String> orgUser) {
     if (!InMemoryUserAndRoleProvider.PROVIDER_NAME.equals(user.getProvider())) {
       for (RoleProvider roleProvider : roleProviders) {
         for (Role role : roleProvider.getRolesForUser(user.getUsername())) {
-          roles.add(JaxbRole.fromRole(role));
+          if (Objects.equals(JaxbOrganization.fromOrganization(user.getOrganization()).getId(),
+                  role.getOrganizationId())) {
+            roles.add(JaxbRole.fromRole(role));
+          }
```

Comment:
Wouldn't it be better to check if the provider is responsible for the organization and only call it if he is?

Like it's done a few lines earlier for the user provider:
https://github.com/opencast/opencast/blob/5a493077d1378835b46057001e31e9229ae49b34/modules/userdirectory/src/main/java/org/opencastproject/userdirectory/UserAndRoleDirectoryServiceImpl.java#L306-L310

Calling external services can be expensive, and retrieving data you dismiss afterward is inefficient.

Reply:
Yeah I understand. I added a check for the roleprovider in the same fashion as for the userprovider.